clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1780,22 +1840,26 @@ void PointerVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV,
                                                  std::string &ReasonFailed) {
   PVConstraint *From = dyn_cast<PVConstraint>(FromCV);
   std::vector<Atom *> NewVatoms;
+  std::vector<ConstAtom *> NewSrcAtoms;
   CAtoms CFrom = From->getCvars();
-  CAtoms::iterator I = Vars.begin();
-  CAtoms::iterator J = CFrom.begin();
   if (CFrom.size() != Vars.size()) {
     ReasonFailed = "transplanting between pointers with different depths";
     return;
   }
-  while (I != Vars.end()) {
-    Atom *IAt = *I;
-    Atom *JAt = *J;
+  for (unsigned AtomIdx = 0; AtomIdx < Vars.size(); AtomIdx++) {
+    Atom *IAt = Vars[AtomIdx];
+    Atom *JAt = CFrom[AtomIdx];
     ConstAtom *ICAt = dyn_cast<ConstAtom>(IAt);
     ConstAtom *JCAt = dyn_cast<ConstAtom>(JAt);
     if (JCAt && !ICAt) {
       NewVatoms.push_back(JAt);
+      NewSrcAtoms.push_back(From->SrcVars[AtomIdx]);
     } else {
       NewVatoms.push_back(IAt);
+      if (isa<WildAtom>(From->SrcVars[AtomIdx]))
+        NewSrcAtoms.push_back(SrcVars[AtomIdx]);
+      else
+        NewSrcAtoms.push_back(From->SrcVars[AtomIdx]);
```

Comment:
This logic doesn't match what's above. Shouldn't we keep the original unless it's wild?